app/models/concerns/host_info.rb

```diff
@@ -0,0 +1,24 @@
+module HostInfo
+  extend ActiveSupport::Concern
+
+  def info
+    info_hash = {}
+
+    self.class.info_providers.each do |provider_method|
```

Comment:
Neat, I suppose this is meant for plugins to add info to the ENC through `register_info_provider`. Can you write a bit of docs about these methods? Something like https://github.com/theforeman/foreman/blob/develop/app/services/foreman/parameter_filter.rb - just a way for people to know what this interface is for